buildSrc/src/main/java/org/springframework/boot/build/classpath/CheckClasspathForProhibitedDependencies.java

```diff
@@ -79,6 +79,9 @@ private boolean prohibited(ModuleVersionIdentifier id) {
 		if (group.equals("javax.money")) {
 			return false;
 		}
+		if (group.equals("javax.xml.bind") && id.getName().equals("jaxb-api")) {
```

Comment:
The Apache Pulsar client lib unfortunately brings this unfortunate baggage w/ it. The Pulsar admin client relies on this as it still codes to the old `javax.*` API. 

I have raised this https://github.com/apache/pulsar/issues/19856 w/ the Pulsar team and it will be address in the `3.0.0` release (they are going straight from `2.11.0 -> 3.0.0` so its not too far out on timeline but I'm not 100% certain the expected date). 

Would it be possible to augment the plugin to allow this dep. only in Pulsar (until 3.0.0 is out)?

Reply:
I don't think we can do that. Even in 2.x we prohibit dependencies from a `javax.*` group in favor of the more modern `jakarta.*` equivalents that use the `javax` namespace for their packages. Even if we tolerated the `javax.xml.bind` API dependencies, I think we'd quickly hit problems with the runtime which uses the same coordinates (`org.glassfish.jaxb:jaxb-runtime`) for its `javax` and `jakarta` implementations.

Reply:
> I think we'd quickly hit problems with the runtime which uses the same coordinates (org.glassfish.jaxb:jaxb-runtime) for its javax and jakarta implementations.

I don't see `org.glassfish.jaxb:jaxb-runtime` using any `javax` classes:
* https://mvnrepository.com/artifact/org.glassfish.jaxb/jaxb-bom/4.0.2
  * https://mvnrepository.com/artifact/org.glassfish.jaxb/jaxb-runtime/4.0.2
    * https://mvnrepository.com/artifact/org.glassfish.jaxb/jaxb-core/4.0.2

I don't think that will happen as the classes are "isolated" from one another by package name (`javax` / `jakarta`). In effect, the only thing still looking at the `javax` would be the older Pulsar client admin.

Reply:
> the javax classes and the runtime for the jakarta classes on the classpath at the same time due to them have the same Maven coordinates.

The runtime has the same coordinates and only a version difference of that coordinates dictates which version it uses (javax/jakarta), is that correct? 

> The API without a runtime to go with it is of limited or perhaps even no use and I don't want users to have to try to straighten out that mess.

Agreed - that would not be good. 
Although, for the Pulsar client, it does shade the older runtime that it uses (it unfortunately did not shade the API). 
So users would not need to enable the older runtime - I think it would be invisible to them. 

I will setup an example app and see what happens.

Reply:
> The runtime has the same coordinates and only a version difference of that coordinates dictates which version it uses (javax/jakarta), is that correct?

Correct. 4.0.x is `jakarta.*`.

Reply:
Good news, the updates to the Pulsar client were merged and will make it into the `3.0.0-RC2` release. I will update the core to use that version and we should be rid of this violation.

Reply:
We are now using Pulsar `3.0.0` and this is no longer an issue.